static/app/views/dashboards/editAccessSelector.tsx

```diff
@@ -222,8 +227,8 @@ function EditAccessSelector({dashboard, onChangeEditAccess}: EditAccessSelectorP
       onOpenChange={() => {
         setMenuOpen(!isMenuOpen);
         if (isMenuOpen) {
-          setSelectedOptions(getSelectedOptions());
           setHasUnsavedChanges(false);
+          setSelectedOptions(getSelectedOptions());
```

Comment:
Do we need to set the selected options when the menu is closed? The state should be handled by the `onSelectOptions` handler so opening and closing shouldn't have an effect on that